cli/azd/pkg/input/console.go

```diff
@@ -591,31 +580,79 @@ func (c *AskerConsole) Handles() ConsoleHandles {
 	return c.handles
 }
 
-const cMinConsoleWidth int = 40
-
 func getConsoleWidth() int {
 	width, _ := consolesize.GetConsoleSize()
-	if width < cMinConsoleWidth {
-		return cMinConsoleWidth
-	}
 	return width
 }
 
+func (c *AskerConsole) handleResize(width int) {
+	c.consoleWidth.Store(int32(width))
+
+	c.spinnerLineMu.Lock()
+	if c.spinner.Status() == yacspin.SpinnerRunning {
+		line := c.spinnerLine(c.spinnerCurrentTitle, c.currentIndent.Load())
+		c.spinner.Message(line.Message)
+		_ = c.spinner.CharSet(line.CharSet)
+		c.spinner.Prefix(line.Prefix)
+	}
+	c.spinnerLineMu.Unlock()
```

Comment:
I feel like this `Console` component is getting a bit out of hand maintaining state, mutexes, etc.  It would be nice if we come up with a plan to wrap our common components like the spinner into its own package/component that is easier to use without all of the orchestration involved like we have today.

Reply:
That was my initial thought as well. The tradeoff is though we lose flexibility if we decide to encapsulate this further.

For example, to wrap the spinner and to properly encapsulate, we'd then need to decide what and what not to show. This ends up being cumbersome when we do want to access underlying `yacspin.Spinner` functionality.